engine/src/main/java/io/zeebe/engine/processing/job/JobThrowErrorProcessor.java

```diff
@@ -35,12 +58,60 @@ private void acceptCommand(
       final TypedRecord<JobRecord> command, final CommandControl<JobRecord> commandControl) {
     final long jobKey = command.getKey();
 
-    final JobRecord job = state.getJob(jobKey);
+    final JobRecord job = jobState.getJob(jobKey);
     job.setErrorCode(command.getValue().getErrorCodeBuffer());
     job.setErrorMessage(command.getValue().getErrorMessageBuffer());
 
-    state.throwError(jobKey, job);
-
     commandControl.accept(JobIntent.ERROR_THROWN, job);
   }
+
+  @Override
+  public void afterAccept(
+      final TypedCommandWriter commandWriter,
+      final StateWriter stateWriter,
+      final long jobKey,
+      final Intent intent,
+      final JobRecord job) {
+
+    final var serviceTaskInstanceKey = job.getElementInstanceKey();
+    final var serviceTaskInstance = elementInstanceState.getInstance(serviceTaskInstanceKey);
+
+    if (serviceTaskInstance != null && serviceTaskInstance.isActive()) {
+
+      final var errorCode = job.getErrorCodeBuffer();
+
+      final boolean errorThrown =
+          errorEventHandler.throwErrorEvent(errorCode, serviceTaskInstance, stateWriter);
```

Comment:
Thanks for spotting it.

I did look at it, but this will be a major change. I am not even sure what is supposed to happen in the new workflow engine. 
I asked on Slack https://camunda.slack.com/archives/C01PCGQK2H2/p1614342051015200 but so far no resolution.

Personally, I would prefer to tackle this bit in a separate issue and not expand the scope of this PR. What do you think?

Reply:
After discussing it with Phil he proposed a different approach:
* reject the throw error command if no catch event can be found
* Move trigger event logic into event applier
* Extend tests to cover these cases